core/src/main/java/org/axonframework/config/SagaConfiguration.java

```diff
@@ -377,8 +508,44 @@ public EventProcessor getProcessor() {
         return sagaManager.get();
     }
 
-    @Override
-    public void shutdown() {
-        processor.get().shutDown();
+    private static String eventProcessorName(Class<?> sagaType) {
+        return AnnotationUtils.findAnnotationAttributes(sagaType, ProcessingGroup.class)
+                              .map(attrs -> (String) attrs.get("processingGroup"))
+                              .orElse(sagaType.getSimpleName() + "Processor");
+    }
+
+    private EventProcessorRegistry eventProcessorRegistry() {
+        return config.eventProcessorRegistry();
     }
+
+    private static class ProcessorInfo {
```

Comment:
This class is needed to decide whether to create a new `EventProcessor` to add it to an existing one, right?

Reply:
Exactly!